### DISCUSSION TITLE
Concurrent checkpoint creation leads to corrupt delta table for delta-rs readers

### DISCUSSION THREAD

**Discussion by Werepyrex10** [delta-io/delta-rs#3244] [Q&A] [answered]
I have the following setup:
- One process which is responsible with creating checkpoint for a delta table, on a specific frequency. This application is using RUST delta-rs library version: `0.17.3`
- One databricks notebook which runs an `optimize` job on top of the delta table once a day

When both processes decide to create a checkpoint on the same version, there is no failure on writing, since the notebook does a multi-part checkpoint, while the delta-rs process does a single-part checkpoint.
Here is a preview of the result of both operations, as seen in the delta logs:
![image](https://github.com/delta-io/delta-rs/assets/8067498/7e313481-3c27-4315-be0c-9afa3bca21a1)

After this occurs, when trying to open the table with the delta-rs lib, we get the following error:
```
Failed to create delta ops: MetadataError("Number of checkpoint files '3' is not equal to number of checkpoint metadata parts 'None'")
```

This is because of the way the library counts the number of parts
https://github.com/delta-io/delta-rs/blob/rust-v0.17.3/crates/core/src/kernel/snapshot/log_segment.rs#L447-L452

Should the library ignore the multi-part files if the `_last_checkpoint` file does not have any `parts` specified ?

**Comment by djouallah** [comment, marked as answer]
I don't thinking using multiple delta writers in the same table is a good idea, the whole ecosystem is not mature enough, just use one writer for everything.

**Comment by ion-elgreco** [comment]
@Werepyrex10 I suggest you disable the checkpointing in delta-spark or delta-rs for now.

**Comment by rtyler** [comment]
@Werepyrex10 What storage backend is this? If it's S3, is the Databricks cluster using the same S3DynamoDbLogStore configuration as the delta-rs process?

**Comment by Werepyrex10** [comment]
Hey @rtyler , we are using azure blob storage as the storage backend

**Comment by mriccardi89** [comment]
We had the same issue, or you config a Dynamo for the delta log or you use only 1 writer. We ended up with the 1 writer solution.

**Comment by one-melange** [comment]
Hello, I am trying to read a delta table that someone else created, and I am getting the error

`_internal.DeltaError: Table metadata is invalid: Number of checkpoint files '0' is not equal to number of checkpoint metadata parts 'None'`

Is there nothing I can do from my side?

**Comment by chiragjn** [comment]
It has been a while since 0.17.x, is it now considered safe to have more than once delta-rs process writing to the same table?
